### ISSUE TITLE
Cross platform support

### ISSUE THREAD

**Issue by Fuzzillogic** [SpectralVectors/PythonPackager#1]
The [current code](https://github.com/SpectralVectors/PythonPackager/blob/main/__init__.py#L43)  tells PIP to install the packages in a Windows-specific location, which of course won't work on e.g. Linux. The same issue exists in the [PosePipe script](https://github.com/SpectralVectors/PosePipe). I'm not sure if --target is even needed, but according to comments on [B3B on installing packages in Blender](https://b3d.interplanety.org/en/how-to-install-required-packages-to-the-blender-python-with-pip/) it might conflict with packages already installed on the system. 

For PosePipe, I had success with using `f"--prefix={sys.prefix}"` instead of the Windows-specific `--target`. I use Blender 3.2.1 (from the official .tar.gz) on Kubuntu 22.04. After this change the packages installed correct, and I got the tracker working (up until a crash when stopping the capture with Esc, but that seems to be a problem elsewhere.)

Still, I guess there are still problems with Blender installs installed using (system wide) package managers, where users aren't allowed to add stuff to Blender's Python libraries. I don't know what to do about that.

**Comment by SpectralVectors** [comment]
Yes, there are issue with cross platform support, I've only started using Linux in the past month, and I'm finding that a few of my addons don't work the same way.

In this case, it was intentional, AFAIK Blender on Linux will use the system Python install, whereas on Windows it will use a separate executable that comes prepackaged with the Blender installer .msi, I kept trying to pip install packages, but the terminal would say, "can't write to this location, defaulting to system install" or something, and I would never be able to access the packages inside of Blender.

There is also a separate issue that I realized: some packages will install to a 'site-packages' folder within the Python folder, and even if you're directing everything to the right Python, it still won't see the packages.

Most of this will be addressed in an update to the addon, I'm a big fan of B3D Interplanetary, and Nikita actually posted an updated version of the link you sent that I think fixes our issues: https://b3d.interplanety.org/en/installing-python-packages-with-pip-in-blender-on-windows-10/

When I get a chance I'm going to update the function to be more like that script, as it seems to work for me. I tested scipy with my script and his, and his worked. I know it says Windows 10, but looking at the code it seems pretty platform independent, or at least worth giving a shot on Linux to see if it just works out of the box.

**Comment by Fuzzillogic** [comment]
> In this case, it was intentional, AFAIK Blender on Linux will use the system Python install

Well, it's Linux, so: _it depends..._ Installing Blender from the default Ubuntu repositories would indeed use the system's Python install, if I read the dependencies correctly. But that would give only a older version of Blender. The official .tar.gz file you can download from Blender's website has its own Python interpreter, which is separate from the rest of the system. That's what I used, and what works.

I've looked into the flatpak of Blender as well, and it too carries its own Python. However, even when installing the Blender flatpak in --user mode (thus the binaries are placed in your home dir) I couldn't get it to work. I _guess_ that flatpak still prevents writing to the application's  directory.

**Comment by SpectralVectors** [comment]
OK, I'm running Garuda, and I'm still getting to grips with the Linux file system so I'm not even sure which is the case with my system!

The subprocess module just runs terminal commands, so you could try a 'sudo' in the install line, but you might need to then add a separate block to input your password. Not sure if that'll work, but it's something

**Comment by Fuzzillogic** [comment]
Using `sudo` inside add-ons won't work: you won't get a popup or something like that. But requiring super user rights would be a bad idea regardless, on Linux and on Windows.

However, there's an [open task on the developer.blender.org on installing modules in add-ons](https://developer.blender.org/T71420), and [Roman Markov's comment](https://developer.blender.org/T71420#1116574) has a better and working solution.

In fact, I got it to work with PosePipe. After adding the `ensure_site_packages` function, I simply replaced the current `install` with this:

```
def install():
    """ Install MediaPipe and dependencies behind the scenes """
           
    ensure_site_packages([
        ("cv2", "opencv-python"),
        ("mediapipe","mediapipe"),
    ])
```
That's all! (protobuf isn't needed at all, it seems)

Caveats: 
1. on Kubuntu 22.04, with the current official blender-3.2.1-linux-x64.tar.xz, Blender crashes when pressing esc to close the capture window. Issue with libc/mediapipe it seems. But, with Blender 3.2.1 from Flathub it did work fine - although you then need to give the flatpak the extra permission of `device=all` to allow it to access the webcam. You can use flatseal for that as well, also on Flathub.
2. Pip packages then get installed to `~/.local/lib/python3.10/` which is where pip of the host would also install packages. Also, stuff like pip3 got installed in `~/.local/bin`, which takes precedence over the host's pip3 in `/usr/bin`. Ugh. I've tried running Blender inside a venv, didn't help.
 
I'd rather have it installed in the context of the Blender add-on, even if that means that each add-on would have its own copy of packages. It would decouple the add-on packages from the host's packages. But that's something for the Blender devs, I guess.

So ya, it works with this method, but can mess up things in unexpected ways.
